### PR TITLE
Add vision label detection

### PR DESCRIPTION
**Pull request by daspecster**
Based off of #2236.

I was able to DRY up the requests code for the `EntityAnnotation` based detection types.

✅ Waiting for #2236 before rebasing.

### CODE REVIEW

**Comment by daspecster** [comment]
@tseaver LMKWYT!

File: google/cloud/vision/entity.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ def from_api_repr(cls, response):
         :rtype: :class:`~google.cloud.vision.entiy.EntityAnnotation`
         :returns: Instance of ``EntityAnnotation``.
         """
-        bounds = Bounds.from_api_repr(response['boundingPoly'])
+        bounds = []
+        if 'boundingPoly' in response:
+            bounds = Bounds.from_api_repr(response['boundingPoly'])
```
Comment (tseaver) [inline google/cloud/vision/entity.py]: Is this part stray, or a rebase artifact?  It doesn't seem related to the rest of the PR.  Also, I don't see any new tests which exercise the missing `bounds` key case.

Comment (daspecster) [inline google/cloud/vision/entity.py]: So actually there will be more of this in the other PRs for other fields. Some API responses don't have the `boundingPoly` index.  Later on, `locale`, `mid`, `score` are returned optionally as well.

I used `response.get('locale', None)` for those however, so for this I could use 
```python
bounds = Bounds.from_api.repr(response.get('boundingPoly', []))
```

Comment (tseaver) [inline google/cloud/vision/entity.py]: None of the `LABEL_DETECTION_RESPONSE` fixtures supply the `boundingPoly` key, which makes me think it isn't relevant here.  I'm surprised coverage passes, actually, since line 60 won't get exercised.

Comment (daspecster) [inline google/cloud/vision/entity.py]: so I actually just switched [this](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2237/commits/e4d49744450945b28b4711e9469bb4cd0bacedbf) to my suggestion, it seems a lot cleaner this way.

File: google/cloud/vision/entity.py [hunk 2]
```diff
@@ -55,9 +55,7 @@ def from_api_repr(cls, response):
         :rtype: :class:`~google.cloud.vision.entiy.EntityAnnotation`
         :returns: Instance of ``EntityAnnotation``.
         """
-        bounds = []
-        if 'boundingPoly' in response:
-            bounds = Bounds.from_api_repr(response['boundingPoly'])
+        bounds = Bounds.from_api_repr(response.get('boundingPoly', {}))
```
Comment (tseaver) [inline google/cloud/vision/entity.py]: It is desirable to have an "empty" `Bounds` instance in this case?  Wouldn't leaving `bounds` as `None` be a better marker?

File: google/cloud/vision/geometry.py [hunk 3]
```diff
@@ -34,10 +34,8 @@ def from_api_repr(cls, response_vertices):
         :rtype: :class:`~google.cloud.vision.geometry.BoundsBase`
         :returns: Instance of BoundsBase with populated verticies.
         """
-        vertices = []
-        for vertex in response_vertices['vertices']:
-            vertices.append(Vertex(vertex.get('x', None),
-                                   vertex.get('y', None)))
+        vertices = [Vertex(vertex.get('x', None), vertex.get('y', None)) for
+                    vertex in response_vertices.get('vertices', [])]
```
Comment (tseaver) [inline google/cloud/vision/geometry.py]: Here, `vertices` is naturally a list, so having an empty one (where the back-end didn't populate it) feels right.

File: google/cloud/vision/entity.py (hunk 2 above)
Comment (daspecster) [inline google/cloud/vision/entity.py]: Yeah I agree! Fixing...

Comment (daspecster) [inline google/cloud/vision/entity.py]: Ok I just added a check in `BoundsBase` to see if the response was `None` that was passed in.

**Comment by daspecster** [comment]
@tseaver anything else?

**Comment by tseaver** [comment]
LGTM pending Travis.

**Comment by daspecster** [comment]
Squashed...going to wait for travis one more time.
